pkg/crc/api/websocket/handler.go

```diff
@@ -0,0 +1,38 @@
+package websocket
+
+type EndpointHandler struct {
+	listeners map[ConnectionListener]struct{}
+	sendData  SendData
+}
+
+type SendData func([]byte)
+
+type ConnectionListener interface {
+	onClientConnected(count int, sendData SendData)
+	onClientDisconnected(count int, sendData SendData)
+}
+
+func NewEndpointHandler(data SendData) *EndpointHandler {
+	handler := &EndpointHandler{
+		listeners: make(map[ConnectionListener]struct{}),
```

Comment:
maybe it would be better if we can avoid using these empty maps with only map keys but an empty `struct`?